forestdb/iterator.go

```diff
@@ -87,5 +88,8 @@ func (i *Iterator) Close() error {
 	if i.curr != nil {
 		i.curr.Close()
 	}
+	if i.parent != nil {
+		i.parent.Close()
```

Comment:
Hmmm, I guess it's poorly named.

One way of thinking about it is that Reader.Close() now kinda means something like "Reader.ReleaseRef()" or "Reader.DecRef()"

And, each iterator instance (with this PR) now owns a ref on its originating Reader, so when an Iterator goes away, it needs to release that ref.

The application, too, has its own ref on the Reader instance, which it releases by calling reader.Close().

So, if the usage patterns ends up like...

r := store.Reader()
// r refs == 1

iter := r.PrefixIterator(...)
// r.refs == 2
...
iter.Close()
// r.refs == 1

r.Close()

Maybe provide another method called decRef() so that intention is more clear?